### PR TITLE
Reputation penalty for sending empty block response

### PR DESCRIPTION
**Pull request by arkpar**
This handles the following scenario:
1. We've learned that a peer has some older fork though block announcement or consensus gossip.
2. We request that block from that peer and they don't send it back.
Currently there's no penalty and the same request will be issues again.

### CODE REVIEW

File: client/network/src/protocol/sync.rs [hunk 1]
```diff
@@ -696,6 +699,10 @@ impl<B: BlockT> ChainSync<B> {
 						}
 						PeerSyncState::DownloadingStale(_) => {
 							peer.state = PeerSyncState::Available;
+							if blocks.is_empty() {
```
Comment (bkchr) [inline client/network/src/protocol/sync.rs:702]: Why are we not doing this for the other states as well?

Comment (arkpar) [reply client/network/src/protocol/sync.rs:702]: When doing full sync it is not that critical, as we'll get new blocks from other peers eventually.

**Comment by gnunicorn** [comment]
bot merge

**Comment by gnunicorn** [comment]
bot merge

**Pull request merged**